in_toto/verifylib.py

```diff
@@ -306,294 +308,582 @@ def verify_all_steps_command_alignment(layout, links_dict):
     verify_command_alignment(command, expected_command)
 
 
-def verify_match_rule(rule, artifact_queue, artifacts, links):
-  """
-  <Purpose>
-    Verifies that target path pattern - 3rd or 5th element of rule - matches
-    at least one file in target artifacts. This might conflict with
-    general understanding of glob patterns (especially "*").
+def _equal_hash_dicts(hash_dict_a, hash_dict_b):
+  """Compare to hash_dict for equality, formats are
+  securesystemslib.formats.HASHDICT_SCHEMA. Return True or False. """
+
+  if set(hash_dict_a.keys()) != set(hash_dict_b.keys()):
+    return False
 
-    Further verifies that each matched target artifact has a corresponding
-    source artifact with a matching hash in the passed artifact dictionary and
-    that this artifact in also the artifact queue.
+  for key in hash_dict_a.keys():
+    if hash_dict_a[key] != hash_dict_b[key]:
```

Comment:
I assume there are no recursive properties, but could we short_circuit to check for equality in both the value and the reference? 

I'm also not sure if this method is [strictly necessary](https://github.com/python/cpython/blob/master/Objects/dictobject.c#L2697-L2737)